client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
+		const post = getSitePost( getState(), action.siteId, action.postId );
+
+		let message;
+		if ( post ) {
+			message = translate( 'An error occurred while deleting "%s"', {
+				args: [ truncate( post.title, { length: 24 } ) ]
+			} );
+		} else {
+			message = translate( 'An error occurred while deleting the post' );
+		}
+
+		dispatch( errorNotice( message ) );
+	},
+	[ POST_DELETE_SUCCESS ]: ( action, dispatch ) => {
+		dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
+	},
+	[ POST_SAVE_SUCCESS ]: ( action, dispatch ) => {
+		let text;
+		switch ( action.post.status ) {
+			case 'trash':
+				text = translate( 'Post successfully moved to trash' );
+				break;
+
+			case 'publish':
+				text = translate( 'Post successfully published' );
+				break;
+		}
+
+		if ( text ) {
+			dispatch( successNotice( text ) );
+		}
+	}
```

Comment:
> Isn't this true of any variable? A variable's name is its calling convention

not really. functions defined using `function` and the fat-arrow syntax have a specific way of being invoked: add some parenthesis or use `call()` or `apply()`. What if we call `HANDLERS[ POST_HAS_NO_MORE_CARES_TO_GIVE ]`? What is the expected return of that function call? Well, it's nonsense because that's not a function. The key doesn't exist and thus no handler exists _at that key_, and even more, it's not a function because we didn't happen to set the value at that key _as a function_. The point is that when we have a function by itself, we've got it in full - no assumptions need to be made, no questions remain - it's ours. When we have an object, or a dispatching function of some kind, we have to trust that it's going to give us what we want and that there's no more behavior wrapping around the function call.

Again, this argument is more important not because of this particular bit of code, but in general because we don't have precedent and are trying to figure out what good style looks like in this situation.